plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/tasks/GenerateReportsTask.kt

```diff
@@ -0,0 +1,177 @@
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import org.gradle.workers.WorkerExecutor
+import org.jetbrains.kotlin.konan.file.File
+import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
+import org.jlleitschuh.gradle.ktlint.worker.ConsoleReportWorkAction
+import org.jlleitschuh.gradle.ktlint.worker.GenerateReportsWorkAction
+import org.jlleitschuh.gradle.ktlint.worker.LoadReportersWorkAction
+import java.io.FileInputStream
+import java.io.ObjectInputStream
+import javax.inject.Inject
+
+/**
+ * Generates reports and prints errors into Gradle console.
+ *
+ * This will actually fail the build in case some non-corrected lint issues.
+ */
+@CacheableTask
+abstract class GenerateReportsTask @Inject constructor(
+    private val workerExecutor: WorkerExecutor,
+    projectLayout: ProjectLayout,
+    objectFactory: ObjectFactory
+) : DefaultTask() {
+    @get:Classpath
+    internal abstract val ktLintClasspath: ConfigurableFileCollection
+
+    @get:Classpath
+    internal abstract val reportersClasspath: ConfigurableFileCollection
+
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFile
+    internal abstract val loadedReporterProviders: RegularFileProperty
+
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFile
+    internal abstract val loadedReporters: RegularFileProperty
+
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFile
+    internal abstract val discoveredErrors: RegularFileProperty
+
+    @get:Input
+    internal abstract val reportsName: Property<String>
+
+    @get:Input
+    internal abstract val enabledReporters: SetProperty<ReporterType>
+
+    @get:Input
+    internal abstract val outputToConsole: Property<Boolean>
+
+    @get:Input
+    internal abstract val coloredOutput: Property<Boolean>
+
+    @get:Input
+    internal abstract val outputColorName: Property<String>
+
+    @get:Input
+    internal abstract val ignoreFailures: Property<Boolean>
+
+    @get:Input
+    internal abstract val verbose: Property<Boolean>
+
+    init {
+        // Workaround for https://github.com/gradle/gradle/issues/2919
+        onlyIf {
+            val errorsFile = (it as GenerateReportsTask).discoveredErrors.asFile.get()
+            errorsFile.exists()
+        }
+    }
+
+    /**
+     * Reports output directory.
+     *
+     * Default is "build/reports/ktlint/${taskName}/".
+     */
+    @Suppress("UnstableApiUsage")
+    @get:OutputDirectory
+    val reportsOutputDirectory: DirectoryProperty = objectFactory
+        .directoryProperty()
+        .convention(
+            reportsName.flatMap {
+                projectLayout
+                    .buildDirectory
+                    .dir("reports${File.separator}ktlint${File.separator}$it")
+            }
+        )
+
+    @Suppress("UnstableApiUsage")
+    @TaskAction
+    fun generateReports() {
+        val queue = workerExecutor.classLoaderIsolation { spec ->
```

Comment:
Can you document why you're using classloader isolation here vs process isolation?

Reply:
Fixed via 8d561a1